aviatrix/resource_aviatrix_dcf_mwp_policy_list.go

```diff
@@ -0,0 +1,458 @@
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the policy list.",
+			},
+			"policies": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "List of distributed-firewalling policies.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"action": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{"DENY", "PERMIT", "DEEP_PACKET_INSPECTION_PERMIT", "INTRUSION_DETECTION_PERMIT"}, false),
+							Description: "Action for the specified source and destination Smart Groups. " +
+								"Must be one of DEEP_PACKET_INSPECTION_PERMIT, INTRUSION_DETECTION_PERMIT, PERMIT or DENY.",
+						},
+						"decrypt_policy": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "DECRYPT_UNSPECIFIED",
+							ValidateFunc: validation.StringInSlice([]string{"DECRYPT_UNSPECIFIED", "DECRYPT_ALLOWED", "DECRYPT_NOT_ALLOWED"}, false),
+							Description: "Decryption options for the policy. " +
+								"Must be one of DECRYPT_UNSPECIFIED, DECRYPT_ALLOWED or DECRYPT_NOT_ALLOWED.",
+						},
+						"dst_smart_groups": {
+							Type:        schema.TypeSet,
+							Required:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Description: "Set of destination Smart Group UUIDs for the policy.",
+						},
+						"exclude_sg_orchestration": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+							Description: "If this flag is set to true, this policy will be ignored for SG orchestration.",
+						},
+						"flow_app_requirement": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "APP_UNSPECIFIED",
+							ValidateFunc: validation.StringInSlice([]string{"APP_UNSPECIFIED", "TLS_REQUIRED", "NOT_TLS_REQUIRED"}, false),
+							Description: "Flow application requirement for the policy. " +
+								"Must be one of APP_UNSPECIFIED, TLS_REQUIRED or NOT_TLS_REQUIRED.",
+						},
+						"logging": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+							Description: "Whether to enable logging for the policy.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Name of the policy. Policy names must be unique.",
+						},
+						"port_ranges": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "List of port ranges for the policy.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"lo": {
+										Type:         schema.TypeInt,
+										Required:     true,
+										ValidateFunc: validation.IntAtLeast(0),
+										Description:  "Lower bound of port range.",
+									},
+									"hi": {
+										Type:             schema.TypeInt,
+										Optional:         true,
```

Comment:
Lo seems to be required, but Hi is not, however in the code below where we check if there are port ranges at all, both are required or we fail.